src/components/CityDetails/index.jsx

```diff
@@ -0,0 +1,160 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchCitiesData } from "../ApiData/data";
+import { Link } from "react-router-dom";
+import Modal from "./Modal";
+import "animate.css/animate.min.css";
+import "./styled.scss";
+
+function CityDetails() {
+  const [cityData, setCityData] = useState(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalContent, setModalContent] = useState("");
+  const { index } = useParams();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await fetchCitiesData(index);
+        setCityData(data[index]);
+      } catch (error) {
+        console.error("Ошибка при загрузке данных", error);
+      }
+    };
+
+    fetchData();
+  }, [index]);
+
+  if (!cityData) {
+    return <div>Loading...</div>;
+  }
+
+  const handleModalOpen = (text) => {
+    setModalContent(text);
+    setIsModalOpen(true);
+  };
+
+  const handleModalClose = () => {
+    setIsModalOpen(false);
+  };
+
+  const daterise = new Date(cityData.sys.sunrise * 1000);
+  const formattedRise = daterise.toLocaleTimeString();
+
+  const datesunset = new Date(cityData.sys.sunset * 1000);
+  const formattedSunset = datesunset.toLocaleTimeString();
```

Comment:
To follow DRY, please create a helper function. You will pass an input value, and receive a formatted date string.

Reply:
function formatUnixTimestamp(timestamp) {
  const date = new Date(timestamp * 1000);
  return date.toLocaleTimeString();
}

const formattedRise = formatUnixTimestamp(cityData.sys.sunrise);
const formattedSunset = formatUnixTimestamp(cityData.sys.sunset);

Did I get that right?

Reply:
Yes, you did.